server-side/services/filter-object.service.ts

```
@@ -0,0 +1,54 @@
+import { Client } from "@pepperi-addons/debug-server/dist";
+import { AddonDataScheme } from "@pepperi-addons/papi-sdk";
+import { filterObjectJsonschema, filterObjectSchema, filterObjectSchemaName } from "../schemas-definition";
+import { FilterObject } from "../types";
+import { BasicTableService } from "./basic-table.service";
+
+
+export class FilterObjectService extends BasicTableService<FilterObject>{
+    schemaName: string;
+    schema: AddonDataScheme;
+    jsonSchemaToValidate: any;
+
+    constructor(client: Client, ownerUUID?: string, secretKey?: string) {
+        super(client, ownerUUID, secretKey);
+        this.schemaName = filterObjectSchemaName;
+        this.schema = filterObjectSchema;
+        this.jsonSchemaToValidate = filterObjectJsonschema;
+    }
+
+    async validateReferencedKey(addonData: FilterObject): Promise<boolean> {
+        // validate that referenced key exist
+        if (addonData.PreviousFilter) {
+            const referencedFilterObjectKey = addonData.PreviousFilter;
+            try {
+                const referencedFilterObject = await this.papiClient.addons.data.uuid(this.client.AddonUUID).table(this.schemaName).key(referencedFilterObjectKey).get();
+                if (!referencedFilterObject) {
+                    console.warn(`Reference validation failed for ${this.schemaName} object: ${JSON.stringify(addonData)}\nKey: ${referencedFilterObjectKey} not found.`);
```

failing validations should throw an exception with a description of what failed